content/en/serverless/distributed_tracing/serverless_trace_merging.md

```diff
@@ -0,0 +1,94 @@
+- [Trace merging across AWS Lambda and hosts](#trace-merging-across-aws-lambda-and-hosts)
+
+### Trace merging in a serverless-first environment
+
+AWS X-Ray provides both a backend AWS service (AWS X-Ray active tracing) and a set of client libraries. [Enabling the backend AWS service alone in the Lambda console][2] gives you `Initialization` and `Invocation` spans for your AWS Lambda functions. You can also enable AWS X-Ray active tracing from the API Gateway and Step Function consoles.
+
+Both the AWS X-Ray SDK and Datadog APM client libraries (`dd-trace`) add medatada and spans for downstream calls by accessing the function directly. Assuming you are using `dd-trace` to trace at the handler level, your setup should be similar to the following:
+
+1. You have enabled [AWS X-Ray active tracing][2] on your Lambda functions from the AWS Lambda console and our [AWS X-Ray integration within Datadog][3].
+2. You have instrumented your Lambda functions with Datadog APM (`dd-trace`) by following the [installation instructions for your Lambda runtime][4].
+3. The AWS X-Ray client libraries should not be installed on your Lambda functions. Otherwise, you will see duplicate traces.
+
+To merge the traces, set the corresponding environment variable as outlined below:
+
+{{< tabs >}}
+{{% tab "Node.js" %}}
+```javascript
+module.exports.hello = datadog(
+    (event, context, callback) => {
+        longCalculation();
+
+        callback(null, {
+            statusCode: 200,
+            body: 'Hello from serverless!'
+        });
+    },
+    { mergeDatadogXrayTraces: true }
+);
+```
+{{% /tab %}}
+
+{{% tab "Python" %}}
+
+Set the `DD_MERGE_XRAY_TRACES` environment variable to `True` on your Lambda function.
+
+{{% /tab %}}
+
+{{% tab "Ruby" %}}
+
+Set the `DD_MERGE_DATADOG_XRAY_TRACES` environment variable to `True` on your Lambda function.
+
+{{% /tab %}}
+
+{{< /tabs >}}
+
+### Tracing across AWS Lambda and hosts
+
+If you have installed Datadog's tracing libraries (`dd-trace`) on both your Lambda functions and hosts, your traces will automatically show you the complete picture of requests that cross infrastructure boundaries, whether it be AWS Lambda, containers, on-prem hosts, or managed services. 
+
+If `dd-trace` is installed on your hosts with the Datadog Agent, and your serverless functions are traced with AWS X-Ray, your setup should be similar to the following:
+
+1. You have installed the [AWS X-Ray integration][2] for tracing your Lambda functions, enabling both AWS X-Ray active tracing and installing the X-Ray client libraries.
+2. You have installed the [Datadog Lambda Library for your Lambda runtime][4], and the `DD_TRACE_ENABLED` environment variable is set to `false`.
+3. [Datadog APM][5] is configured on your hosts and container-based infrastructure.
+
+Then, for X-Ray and Datadog APM traces to appear in the same flame graph, all services must have the same `env` tag.
```

Comment:
Is this accurate, or do you need to set the above environment variable for this too? @tianchu

Reply:
I don't think this is a hard requirement, though I never tried it specifically.